src/pages/instances/InstanceSearchFilter.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { FC } from "react";
+import { LxdInstance, LxdInstanceStatus } from "types/instance";
+import {
+  InstanceFilters,
+  enrichStatuses,
+  instanceStatuses,
+  instanceTypes,
+} from "util/instanceFilter";
+import { SearchAndFilter } from "@canonical/react-components";
+import {
+  SearchAndFilterData,
+  SearchAndFilterChip,
+} from "@canonical/react-components/dist/components/SearchAndFilter/types";
+
+interface Props {
+  instances: LxdInstance[];
+  setFilters: (newFilters: InstanceFilters) => void;
+}
+
+const InstanceSearchFilter: FC<Props> = ({ instances, setFilters }) => {
+  const profileSet = [
+    ...new Set(instances.flatMap((instance) => instance.profiles)),
```

Comment:
I think we should display all profiles in the project, not only the used ones. One application will be to link here (with that profile filter set) from a profile page. If the profile is not available (because no instances use it) that application might not work.

Reply:
The application that you mentioned will exist only for profiles that have at least 1 instance, not for the "empty" ones. There is an empty state for them, so they won't have any link, and hence there's no need to pollute the filter panel with all the empty profiles.